### PR TITLE
Clean up and improve error message for KyuubiBaseResultSet

### PR DESCRIPTION
**Pull request by pan3793**
### Why are the changes needed?

Backport https://github.com/apache/hive/commit/2b0e424daa67bfa62e695701533fb80b8f10f383

The commit does not have a Hive ticket.

### How was this patch tested?

Pass GHA to ensure it breaks nothing, and verify locally that the reported error message does contain the data when failing to parse the illegal date.

image

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiBaseResultSet.java [hunk 1]
```diff
@@ -159,23 +200,23 @@ public int getConcurrency() throws SQLException {
...
-      if (obj instanceof String) {
+    if (obj instanceof String) {
+      try {
         return Date.valueOf((String) obj);
+      } catch (Exception e) {
+        throw new SQLException(
+            "Illegal conversion to Date from column " + columnIndex + " [" + obj + "]", e);
       }
-    } catch (Exception e) {
-      throw new KyuubiSQLException("Cannot convert column " + columnIndex + " to date: " + e, e);
```
Comment (pan3793) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiBaseResultSet.java:174]: it's horrible that the previously reported error message does not contain the illegal data!

**Comment by pan3793** [comment]
scala 2.13 test failure is irrelevant

**Review by turboFei — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merged to master/1.10.1/1.9.3

**Comment by bowenliang123** [comment]
👍

**Pull request closed without merging**